bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -227,6 +247,16 @@ protected void removeAddonService(AddonService featureService) {
         this.addonServices.remove(featureService);
     }
 
+    @Override
+    public void onChanged(List<CidrAddress> added, List<CidrAddress> removed) {
+        // Nothing to do
+    }
+
+    @Override
+    public void onPrimaryAddressChanged(@Nullable String oldPrimaryAddress, @Nullable String newPrimaryAddress) {
+        startScan();
```

Comment:
This will start the scan with a 20s delay again. This delay should not be there for the second scan, only for the original one.
When testing with a modified setup-wizard, I intended to allow 10s for a rescan after a network change. Keeping to 20s delay in the `startScan` method would force using a 25s delay after network change in the setup wizard.

Reply:
Right, I have removed the delay for the primary address changed event.

Reply:
Btw, this might have been the reason for my first test mentioned here failing: https://github.com/openhab/openhab-core/pull/4036#issuecomment-2053595358 (due to 20s delay and impatience)